### ISSUE TITLE
Docker process is being run under root

### ISSUE THREAD

**Issue by bukajsytlos** [NodeBB/NodeBB#9888]
tl;dr
Docker process should be run as unprivilleged user "node"

Long story:
First of all, I am not nodejs developer. Maybe there is other solution.
So, we are mounting node_modules as a volume to make startups faster. (related to https://github.com/NodeBB/NodeBB/issues/6426)
Since 1.18 docker images, we have problem with this solution as npm install will fail with lot of ENOENT.
I guess sometime around this period, node:lts switched from 12 -> 14 (in my opinion not using stable tags is also an issue).
It works if I build nodebb 1.18.4 from node:12.
For node:14 I had to switch everything to be run under unprivilleged user "node".

Since this looks like a best practice anyway according to 

https://snyk.io/blog/10-best-practices-to-containerize-nodejs-web-applications-with-docker/
https://github.com/goldbergyoni/nodebestpractices/blob/master/sections/security/non-root-user.md

would you accept change to dockerfile to create app under "node" user?

Thanks

**Comment by julianlam** [comment]
Yes, no objections to that.

**Comment by katywings** [comment]
Just for your info: In some environments running the container via a different user can result in permission problems. E.g. In my case I am using podman, which already has a build-in mechanism to run containers rootless. The only way how I was able to get the NodeBB docker-compose to run, was to remove all `USER node` and `chown -R node:node` from the `Dockerfile` 😅.
